tracer/src/Datadog.Trace.SourceGenerators/Helpers/StringArrayComparer.cs

```diff
@@ -0,0 +1,53 @@
+// <copyright file="StringArrayComparer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using Datadog.Trace.SourceGenerators.Helpers;
+
+namespace Helpers
+{
+    internal class StringArrayComparer : IComparer<EquatableArray<string>>
```

Comment:
Ah, I guess this is because you specifically want `Ordinal` comparison? Should we extract the `StringComparison` type to a field to make it extensible for the future, e.g. if we want to ignore case changes?